File: frontend/types/models/simpleRecord.ts
export type SimpleRecordStatus = "draft" | "active" | "published" | "closed" | "archived";

export type SimpleRecord = {
  id: string;
  module: string;
  title: string;
  category: string;
  summary: string;
  status: SimpleRecordStatus;
  record_date: string;
  contact_name: string | null;
  contact_email: string | null;
  amount: number | null;
  location: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
};

export type SimpleRecordFormValues = {
  title: string;
  category: string;
  summary: string;
  status: SimpleRecordStatus;
  record_date: string;
  contact_name: string;
  contact_email: string;
  amount: string;
  location: string;
  notes: string;
};

export type SimpleRecordListResponse = {
  items: SimpleRecord[];
  total: number;
  active: number;
  published: number;
  archived: number;
};

File: backend/app/models/simple_record.py
from __future__ import annotations

import uuid
from datetime import date, datetime
from decimal import Decimal

from sqlalchemy import Date, DateTime, ForeignKey, Numeric, String, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class SimpleRecordModel(Base):
    __tablename__ = "simple_records"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    organization_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("organizations.id", ondelete="RESTRICT"),
        nullable=False,
        index=True,
    )
    module: Mapped[str] = mapped_column(String(40), nullable=False)
    title: Mapped[str] = mapped_column(String(180), nullable=False)
    category: Mapped[str] = mapped_column(String(80), nullable=False, default="general")
    summary: Mapped[str] = mapped_column(String(1200), nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="active")
    record_date: Mapped[date] = mapped_column(Date, nullable=False)
    contact_name: Mapped[str | None] = mapped_column(String(140), nullable=True)
    contact_email: Mapped[str | None] = mapped_column(String(255), nullable=True)
    amount: Mapped[Decimal | None] = mapped_column(Numeric(12, 2), nullable=True)
    location: Mapped[str | None] = mapped_column(String(180), nullable=True)
    notes: Mapped[str | None] = mapped_column(String(1200), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False
    )
